### ISSUE TITLE
False positive for /i regex modifyer

### ISSUE THREAD

**Issue by b1rdex** [kalessil/phpinspectionsea#83]
```php
$cyrillicLayoutRegex = '/[а-яё]/iu';
```

Suggests to remove `/i` because can't find `a-z`.

**Comment by kalessil** [comment]
Thank you for reporting @b1rdex  , this is fixed now.

PS: I pushed a fresh binary as well.

**Comment by funivan** [comment]
@kalessil this is wrong. If we have `u` then we should leave `i`
```php
    $data = [
        'а',
        'А',
        'я',
        'Я',
        'Б',
        'б',
        'ї',
      ];
      foreach ($data as $value) {
        preg_match('/[а-яё]/iu', $value, $result);
        echo $value . ' = ' . ($result[0] ?? '');
        echo "\n";
      }
```
Result: 
```
а = а
А = А
я = я
Я = Я
Б = Б
б = б
ї = 
```
So `i` works fine with `u` =)

**Comment by kalessil** [comment]
I'm not sure what exactly do you mean @funivan, can you elaborate a little bit more - which part of the fix you are referencing to?

**Comment by funivan** [comment]
If we have regexp `$cyrillicLayoutRegex = '/[а-яё]/iu';`
Plugin will suggest us to remove modifier `i`.  As I can see message will be the following: `'i' modifier is ambiguous here (no alphabet characters in given pattern)`
But this logic is wrong. If we remove `i` our regexp can change behaviour. 
Regexp `/[а-яё]/iu` can match cyrillic lower case and upper case letters.

**Comment by funivan** [comment]
@kalessil sorry. I misunderstood comment of the b1rdex This feature is fixed. Thanks for the tests. According to them I figure out that I was wrong

**Comment by kalessil** [comment]
No worries, I was just wondering if missed something. 
Now it twice checked so the fix looks good as for me ).

**Comment by funivan** [comment]
One more thing. This will be hard to check but...
Our regexp is  `'/[а-яё]/i'`
There are two cases
 - In this case `i` is ambiguous 
 - We can suggest to add `u`

But definitely programmer should be noticed to do something. =)

**Comment by dryabov** [comment]
It is really good idea to detect strings with non-ASCII (>127) characters and suggest to add `u` modifier, because of it is quite frequent bug with copy-paste regexps from other languages (e.g. JavaScript).

**Comment by kalessil** [comment]
@funivan, @dryabov : thatk you for an idea, implemented and [avaialable](https://github.com/kalessil/phpinspectionsea/commit/721e247f3ce3ced68652283ec2e6d15a59d7ec2c) now (fresh binary pushed).

**Comment by funivan** [comment]
@kalessil :+1: I will test it next week
